### PR TITLE
fix(experience): routing with pluggable layout

### PR DESCRIPTION
**Pull request by dunqan**
Fixes routing issues for SSR

closes: [HRZ-90790](https://spryker.atlassian.net/browse/HRZ-90790)

[HRZ-90790]: https://spryker.atlassian.net/browse/HRZ-90790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/experience/src/mixins/layout.mixin.ts [hunk 1]
```diff
@@ -196,26 +195,31 @@ export const LayoutMixin = <T extends Type<LitElement & LayoutAttributes>>(
           options: this.$options() as CompositionProperties,
         },
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/mixins/layout.mixin.ts]: no-non-null-assertion is no longer needed

File: libs/platform/experience/src/mixins/layout.mixin.ts [hunk 2]
```diff
@@ -196,26 +195,31 @@ export const LayoutMixin = <T extends Type<LitElement & LayoutAttributes>>(
...
+      undefined as
+        | {
+            template: TemplateResult;
+            experience?: LayoutMixinRender['experience'];
+          }
+        | undefined,
+      {
+        equal: (a, b) => JSON.stringify(a) === JSON.stringify(b),
+      }
+    );
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/mixins/layout.mixin.ts:211]: Took me a while to understand this one. We could move the typing into the generic.

**Review by tobi-or-not-tobi — DISMISSED** [review]
LGTM, but with 2 small nitpicks

File: libs/platform/experience/src/mixins/layout.mixin.ts (hunk 2 above)
Comment (dunqan) [reply libs/platform/experience/src/mixins/layout.mixin.ts:211]: I have no idea why, but it didn't work :( and I was trying different approaches.

Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/mixins/layout.mixin.ts:211]: I had no issues with it, doing:

```ts
protected $template = signal<
      | {
          template: TemplateResult;
          experience?: LayoutMixinRender['experience'];
        }
      | undefined
    >(undefined, {
      equal: (a, b) => JSON.stringify(a) === JSON.stringify(b),
    });
```

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**